src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
```

Comment:
> this gives a tiny bit of perf improvement

There is no perf improvement. Any decent compiler generates exact same code for both `delete` and `delete []` on primitive types. It is just that the static analysis tools tend to complain when you mix `new []` and `delete`. We should just fix it.